File: Library.Core.Web/src/app/view-models/author/getAuthorView.ts
export class GetAuthorView {
  constructor(
    public authors: AuthorGetAuthorViewItem[],
    public allBooks: BookGetAuthorViewItem[]
  ) { }
}

export class AuthorGetAuthorViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public birthday: number = 0,
    public deathday: number = 0,
    public books?: BookGetAuthorViewItem[]
  ) { }
}

export class BookGetAuthorViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/Author/GetAuthorView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Author
{
    public class GetAuthorView
    {
        public ICollection<AuthorGetAuthorViewItem> Authors { get; set; }
        public ICollection<BookGetAuthorViewItem> AllBooks { get; set; }

        public GetAuthorView()
        {
            Authors = new List<AuthorGetAuthorViewItem>();
            AllBooks = new List<BookGetAuthorViewItem>();
        }
    }

    public class AuthorGetAuthorViewItem
    {
        public int AuthorId { get; set; }
        public string Name { get; set; }
        public int Birthday { get; set; }
        public int Deathday { get; set; }

        public ICollection<BookGetAuthorViewItem> Books { get; set; }

        public AuthorGetAuthorViewItem()
        {
            Books = new List<BookGetAuthorViewItem>();
        }
    }

    public class BookGetAuthorViewItem
    {
        public int BookId { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookGetAuthorViewItem()
        {

        }
    }
}
